### ISSUE TITLE
[BAD] False-positive "stop-iteration-return" in nested generators

### ISSUE THREAD

**Issue by ArcherOfFaith** [PyCQA/pylint#3770]
For the case of generator call inside of generator, such this
```
def foo(gen):
    res = next(gen)
    yield res
```
there is no proper way to fix "stop-iteration-return" error, caused by second row

**Comment by AWhetter** [comment]
You should catch the `StopIterator` exception from the call to `next()` and `return` when the exception is caught. See [PEP-479](https://www.python.org/dev/peps/pep-0479/) for more on why this warning exists.
